kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/apt/metadata.kt

```diff
@@ -510,15 +514,34 @@ private fun List<AnnotationSpec>?.qualifiers(
 
 private fun List<AnnotationSpec>?.jsonName(): String? {
   if (this == null) return null
-  val name = find { it.typeName == JSON }?.let { annotation ->
-    val mirror = requireNotNull(annotation.tag<AnnotationMirror>()) {
-      "Could not get the annotation mirror from the annotation spec"
-    }
-    mirror.elementValues.entries.single {
-      it.key.simpleName.contentEquals("name")
-    }.value.value as String
+  return filter { it.typeName == JSON }.firstNotNullOfOrNull { annotation ->
+    annotation.jsonName()
+  }
+}
+
+private fun List<AnnotationSpec>?.jsonIgnore(): Boolean {
+  if (this == null) return false
+  return filter { it.typeName == JSON }.firstNotNullOfOrNull { annotation ->
+    annotation.jsonIgnore()
+  } ?: false
+}
+
+private fun AnnotationSpec.jsonName(): String? {
+  return elementValue<String>("name").takeUnless { it == Json.UNSET_NAME }
+}
+
+private fun AnnotationSpec.jsonIgnore(): Boolean {
+  return elementValue<Boolean>("ignore") ?: false
+}
+
+private fun <T> AnnotationSpec.elementValue(name: String): T? {
+  val mirror = requireNotNull(tag<AnnotationMirror>()) {
+    "Could not get the annotation mirror from the annotation spec"
```

Comment:
Should we print any diagnostics data here to make it easier to triage, or is it a "this should never happen" case?

Reply:
The latter!